### PR TITLE
#170947563 user can delete his comment

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
add a feature where a user can delete his comment

#### Description of Task to be completed?
- added a controller for deleting a comment
- test the feature
- documented the feature
- internationalized all messages

#### How should this be manually tested?
- clone this repo `https://github.com/andela/devrepublic-bn-backend.git`
- run npm install to install all dependencies
- run the server (npm run reset:dev)
- make comments on a request
- make a delete method to `/api/v1/comments/:commentId`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#170947563](https://www.pivotaltracker.com/story/show/170947563)

#### Screenshots (if appropriate)
Screen Shot 2020-03-16 at 14 31 56

Screen Shot 2020-03-16 at 14 32 05

#### Questions:
- N/A

### CODE REVIEW

File: src/controllers/commentController.js [hunk 1]
```diff
@@ -52,4 +54,27 @@ export default class CommentController {
...
+
+  /**
+   * @param  {object} req
+   * @param  {object} res
+   * @param  {object} next
+   * @return {object} delete comment
+   */
+  static async deleteComment(req, res) {
+    try {
+      const { commentId } = req.body;
```
Comment (izzett222) [inline src/controllers/commentController.js]: hey @Baraka-Mugisha could we use the req.params to contain the commentId rather than putting it in req.body

Comment (Baraka-Mugisha) [reply src/controllers/commentController.js]: That is an interesting idea

File: src/validation/validationRules.js [hunk 2]
```diff
@@ -18,6 +18,17 @@ export const signupInputRules = [
...
 ];
 
+export const loginRules = [
+  check('email').trim().exists()
+    .withMessage('The email is required')
+    .isLength({ min: 1, max: 200 })
+    .withMessage('email can not be empty'),
+  check('password').exists()
+    .withMessage('The password is required')
+    .isLength({ min: 1, max: 200 })
```
Comment (bbaime98) [inline src/validation/validationRules.js]: refer to signup validation rules. password minimum length should be 8, not 1 and also check the max limit.

**Review by bbaime98 — CHANGES_REQUESTED** [review]

File: src/validation/validationRules.js (hunk 2 above)
Comment (Baraka-Mugisha) [reply src/validation/validationRules.js]: It is done

**Review by izzett222 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

**Review by jabichris — COMMENTED** [review]
Nice work @Baraka-Mugisha. But make sure everything you do on this branch is related to the feature.

**Review by izzett222 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

File: src/middlewares/protectRoute.js [hunk 3]
```diff
@@ -179,6 +178,30 @@ export default class protectRoutes {
...
+   * @returns {object} user
+   */
+  static async checkComment(req, res, next) {
+    const { user } = req;
+    const { commentId } = req.params;
+
+    const commentExist = await db.Comments.findOne({
+      where: { id: commentId, deleted: false },
+    });
+    if (!commentExist) {
```
Comment (jabichris) [inline src/middlewares/protectRoute.js]: Good work. But you can refactor this for the sake of clean code.

**Review by bbaime98 — APPROVED** [review]

File: src/middlewares/protectRoute.js [hunk 4]
```diff
@@ -179,6 +178,33 @@ export default class protectRoutes {
...
+
+    const commentExist = await db.Comments.findOne({
+      where: { id: commentId, deleted: false },
+    });
+    // if (!commentExist) {
+    //   return Response.errorResponse(res, 404, res.__('Comment not found'));
+    // }
+    // if (user.id !== commentExist.commentOwner) {
+    //   return Response.success(res, 401, res.__('You are not authorised to delete this comment'));
+    // }
```
Comment (bbaime98) [inline src/middlewares/protectRoute.js]: remove these commented code.

**Review by bbaime98 — CHANGES_REQUESTED** [review]

File: src/middlewares/protectRoute.js (hunk 4 above)
Comment (Baraka-Mugisha) [reply src/middlewares/protectRoute.js]: I removed them

**Review by bbaime98 — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

**Pull request merged**